R/get_output_dir.R

```diff
@@ -0,0 +1,13 @@
+get_output_dir <- function(root, date) {
+  if (date == "today") {
+    date <- format(Sys.Date(), "%Y_%m_%d")
+  }
```

Comment:
I wrote up this function a little while ago when I was working on an output structure for Bobby, but it's more generally useful for R folks working on covid, and it could be used as a check in this function.

```
# test YYYY_MM_DD format
# returns T or F 
test_date <- function(date){

  if(!is.character(date)) return(FALSE)

  if(lengths(regmatches(date, gregexpr('_', date))) != 2) return(FALSE)

  split.date <- strsplit(date, '_')

  if(length(split.date) != 1) return(FALSE)

  split.date <- split.date[[1]]

  if(length(split.date) != 3) return(FALSE)

  if(nchar(split.date[1]) != 4 | nchar(split.date[2]) != 2 | nchar(split.date[3]) != 2) return(FALSE)

  if(sum(is.na(as.numeric(split.date))) != 0) return(FALSE)

  return(TRUE)
}
```

Reply:
would a better regular expression make this easier?

```
> grepl("^\\d{4}_\\d{2}_\\d{2}$", "2020_04_05")
[1] TRUE
> grepl("^\\d{4}_\\d{2}_\\d{2}$", "2020_04_05a")
[1] FALSE   
```

Reply:
yes :)